pkg/ddevapp/typo3.go

```diff
@@ -133,6 +133,10 @@ func setTypo3SiteSettingsPaths(app *DdevApp) {
 		settingsFileBasePath := filepath.Join(app.AppRoot, app.ComposerRoot)
 		settingsFilePath = filepath.Join(settingsFileBasePath, "config", "system", "settings.php")
 		localSettingsFilePath = filepath.Join(settingsFileBasePath, "config", "system", "additional.php")
+	} else if isTypo3Appv12OrHigher(app) {
```

Comment:
I imagine this function needs some renaming to distinguish it from `isTypo3v12OrHigher` - what exactly is the difference with `isTypo3Appv12OrHigher` ? I assume "Typo3App" is somehow different from "Typo3", that should be explained somewhere and probably needs better naming.

Reply:
**isTypo3v12OrHigher**

TYPO3 composer mode, v12 or higher, which relevant because of changed configuration files and paths:

```
public/typo3conf/LocalConfiguration.php  -> config/system/settings.php
public/typo3conf/AdditionalConfiguration -> config/system/additional.php
```

Note that the files are moved out of the docroot (`public/*`) in composer mode.

Suggested rename: : `isTypo3v12OrHigherComposerMode`

Composer mode before TYPO3 v12 installed even in composer mode the public/typo3/sysext/folder ... function used the configured docroot (public/)
and thus handled TYPO3 composer mode before v12 with `isTypo3App`

**isTypo3App**

TYPO3 in non-composer mode (legacy mode) aka symlink based setup. The TYPO3 monorepo (core development) has the "typo3/sysext/*" folder like a legacy setup.

Here a simple check has been used, simply looking that `typo3/` folder exists
in the doc root (TYPO3 core files). 

This check is still correct for non-composer mode and also monorepo (core development) - and kind of worked.

Before TYPO3 v12 the configuration files resides in:

```
public/typo3conf/LocalConfiguration.php
public/typo3conf/AdditionalConfiguration
```

which has changed starting with TYPO3 v12. See `isTypo3Appv12OrHigher` below.

Suggested rename: `isTypo3v11OrLowerComposerAndNonComposerMode`

**isTypo3Appv12OrHigher**

I cloned the **isTypo3App*  function and added the Typo3Version class check to it, which is only available since TYPO3 v12 to use the new placed for configuration files in TYPO3 v12 or newer, not covered by boath methods above:

```
public/typo3conf/LocalConfiguration.php  -> public/typo3conf/system/settings.php
public/typo3conf/AdditionalConfiguration -> public/typo3conf/system/additional.php
```

Suggested rename: isTypo3v12OrHigherLegacyMode


Would the renaming be okay for you ?

Reply:
`isTypo3v12OrHigherLegacyMode()` do you have docs for "legacy mode"? It seems like a reasonable approach in that case, but it needs a link to "legacy mode" docs. 

I have to admit that I thought T3 was always done with composer!

Reply:
The "legacy mode" (actually: "classic" mode) is described here:

https://docs.typo3.org/m/typo3/tutorial-getting-started/main/en-us/Installation/LegacyInstallation.html#legacyinstallation

It is used for core dev installs (also referred to as "monorepo" install):

https://docs.typo3.org/m/typo3/guide-contributionworkflow/main/en-us/Quickstart/Index.html#quickstart

Sorry for that being confusing. For a 26year old system things may get cluttered.

What Stefan is proposing here is backed by the core team, in this matter currently represented by Stefan Bürk, Benjamin Franzke and me too. 😅 (ref: https://typo3.org/community/teams/typo3-development, section "framework mergers").